packages/@aws-cdk/aws-servicecatalog/README2.md

```diff
@@ -0,0 +1,379 @@
+
+---
+
+<!--END STABILITY BANNER-->
+
+[AWS Service Catalog](https://docs.aws.amazon.com/servicecatalog/latest/dg/what-is-service-catalog.html)
+enables organizations to create and manage catalogs of products for their end users that are approved for use on AWS.
+
+## Table Of Contents
+
+- [Portfolio](#portfolio)
+  - [Granting access to a portfolio](#granting-access-to-a-portfolio)
+  - [Sharing a portfolio with another AWS account](#sharing-a-portfolio-with-another-aws-account)
+- [Product](#product)
+  - [Creating a product from a local asset](#creating-a-product-from-local-asset)
+  - [Creating a product from a stack](#creating-a-product-from-a-stack)
+  - [Adding a product to a portfolio](#adding-a-product-to-a-portfolio)
+- [TagOptions](#tag-options)
+- [Constraints](#constraints)
+  - [Tag update constraint](#tag-update-constraint)
+  - [Notify on stack events](#notify-on-stack-events)
+  - [CloudFormation template parameters constraint](#cloudformation-template-parameters-constraint)
+  - [Set launch role](#set-launch-role)
+  - [Deploy with StackSets](#deploy-with-stacksets)
+
+
+The `@aws-cdk/aws-servicecatalog` package contains resources that enable users to automate governance and management of their AWS resources at scale.
+
+```ts nofixture
+import * as servicecatalog from '@aws-cdk/aws-servicecatalog';
+```
+
+## Portfolio
+
+AWS Service Catalog portfolios allow administrators to organize, manage, and distribute cloud resources for their end users.
+Using the CDK, a new portfolio can be created with the `Portfolio` construct:
+
+```ts
+new servicecatalog.Portfolio(this, 'Portfolio', {
+  displayName: 'MyPortfolio',
+  providerName: 'MyTeam',
+});
+```
+
+You can also specify optional metadata properties such as `description` and `messageLanguage`
+to help better catalog and manage your portfolios.
+
+```ts
+new servicecatalog.Portfolio(this, 'Portfolio', {
+  displayName: 'MyFirstPortfolio',
+  providerName: 'SCAdmin',
+  description: 'Portfolio for a project',
+  messageLanguage: servicecatalog.MessageLanguage.EN,
+});
+```
+
+Read more at [Creating and Managing Portfolios](https://docs.aws.amazon.com/servicecatalog/latest/adminguide/catalogs_portfolios.html).
+
+To import an existing portfolio into your CDK application, use the `Portfolio.fromPortfolioArn()` factory method:
+
+```ts
+const portfolio = servicecatalog.Portfolio.fromPortfolioArn(this, 'ImportedPortfolio',
+  'arn:aws:catalog:region:account-id:portfolio/port-abcdefghi');
+```
+
+### Granting access to a portfolio
+
+You can grant access to and manage the `IAM` users, groups, or roles that have access to the products within a portfolio. 
+Entities with granted access will be able to utilize the portfolios resources and products via the console or AWS CLI.
+Once resources are deployed end users will be able to access them via the console or service catalog CLI.
+
+```ts fixture=basic-portfolio
```

Comment:
off topic, but I don't like using these fixtures. Will have a problem with our rosetta extract tool since once this example is extracted it will be hard to tell what a `portfolio` is.

I prefer adding a `declare const portfolio: servicecatalog.Portfolio` rather than using fixtures.

Reply:
Can you point to updated documentation you have for creating readmes?  We added rosetta and set it up like this based on other libraries.

Reply:
the short of it is that our status quo has changed. the updated documentation is [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md#recommendations).

the long story is that we now utilize these README snippets outside the README. They get sourced as example code for APIs in the docs. For example, say your README has this:

To use `foo`, send an s3 bucket into the parameters:

// ```ts fixture=bar.ts-fixture
```ts fixture=bar.ts-fixture
const bucket = new s3.Bucket();
bar.foo(bucket);
```

This code snippet will be used as an eligible example for API documentation on `bar`, `foo`, and also `s3.Bucket`. Say it shows up in `s3.Bucket`. People who read the documentation will have no idea what type `bar` is, since that information is in `bar.ts-fixture` and not visible. That is why we try to write out README snippets like:

// ```ts
```ts
declare const bar: module.Bar;
const bucket = new s3.Bucket();
bar.foo(bucket);
```

That way, no matter where the snippet may end up, we know the type information of `bar`.

Here's an example of this in action: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html#example

As you can see, the example for `s3.Bucket` most likely came from the readme for `codebuild`.

Reply:
This is not a breaking change by any means, so it's non-blocking for GA. But ideally we stick to using `default.ts-fixture` whenever we can, and just use `declare` statements to convey type information for the objects you don't want to fully define.

Reply:
thanks for detailed explanation - I like this more anyway so I will update it